spyder/app/mainwindow.py

```diff
@@ -1236,6 +1237,29 @@ def add_ipm_action(text, path):
         logger.info("*** End of MainWindow setup ***")
         self.is_starting_up = False
 
+    def setup_menus(self):
+        """Hide icon menus in toolbars before displaying them"""
+
+        # Show and hide shortcuts in menus for Mac.
+        # This is a workaround because we can't disable shortcuts
+        # by setting context=Qt.WidgetShortcut there
+        if sys.platform == 'darwin':
+
+            for name in ['file', 'edit', 'search', 'source', 'run', 'debug',

```

Comment:
@juanis2112 now we can use 

`self.menus` so

something like:

`for menu in self.menus`